src/backends/perf_pt/collect.c

```diff
@@ -570,8 +570,9 @@ perf_pt_free_tracer(struct tracer_ctx *tr_ctx) {
     if (tr_ctx->stop_fds[0] != -1) {
         close(tr_ctx->stop_fds[0]);
     }
-    if (tr_ctx->perf_fd) {
+    if (tr_ctx->perf_fd >= 0) {
         close(tr_ctx->perf_fd);
+        tr_ctx->perf_fd = -1;
```

Comment:
Do we know for sure that valid fds have to be `>=` 0?

Reply:
Since open returns -1 on error...

Reply:
Ah, even more to the point the OpenBSD `open` page says "If successful, open() returns a non-negative integer, termed a file descriptor." Just checking!